### PR TITLE
Creating threads with Executor Service

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@jsuchy @felipesere @gemcfadyen @christophgockel

This last PR changes how threads are created. I'm now using the executor service.
All cob spec tests are green now.

### CODE REVIEW

File: src/test/java/de/rabea/server/ExecutorServiceSpy.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    public void execute(Runnable command) {
+        executesHttpServerRunner = true;
+    }
+
+    @Override
+    public void shutdown() {
+        hasShutDown = true;
+
```
Comment (gemcfadyen) [inline src/test/java/de/rabea/server/ExecutorServiceSpy.java:20]: extra line here.

File: src/test/java/de/rabea/server/ExecutorServiceSpy.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public boolean isTerminated() {
+        return false;
+    }
+
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+        return false;
+    }
+
```
Comment (gemcfadyen) [inline src/test/java/de/rabea/server/ExecutorServiceSpy.java:42]: extra line here.

File: src/test/java/de/rabea/server/ExecutorServiceSpyFactory.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    public ExecutorServiceSpyFactory(ExecutorServiceSpy executorServiceSpy) {
+        this.executorServiceSpy = executorServiceSpy;
+    }
+
+    @Override
+    public ExecutorService create(int amount) {
+        return executorServiceSpy;
+    }
+
```
Comment (gemcfadyen) [inline src/test/java/de/rabea/server/ExecutorServiceSpyFactory.java:17]: and here :)

Comment (RabeaGleissner) [inline src/test/java/de/rabea/server/ExecutorServiceSpyFactory.java:17]: Thanks @gemcfadyen 
I eliminated those bad boys on the master branch 😄

**Pull request merged**